internal/edgedbtypes/uuid.go

```diff
@@ -45,6 +46,10 @@ func (id UUID) MarshalText() ([]byte, error) {
 // UnmarshalText unmarshals the id from a string.
 func (id *UUID) UnmarshalText(b []byte) error {
 	s := string(b)
```

Comment:
`s` is always coming from the DB? How can it be malformed?

Reply:
This is for decoding JSON. Who knows where it is coming from. For example, decoding the json body of a post request that you will then use as arguments to an insert.

Reply:
This is never used inside an edgedb query method.

Reply:
I still don't follow. Can you come up with an example?

Reply:
You get a POST request with a json payload. One or more of the values in that payload are UUIDs that you want to then use in a query. The json package uses the [TextUnmarshaler interfaces](https://golang.org/pkg/encoding/#TextUnmarshaler) to decode to types it doesn't know about. 

```go
type Movie struct {
    Title string
    Director struct {
        ID edgedb.UUID
    }
}

// CreateMovie is a request handler
func CreateMovie(c *gin.Context) {
    var payload Movie
    err := json.NewDecoder(c.Request.Body).Decode(&payload)
    if err != err { 
        panic(err) 
    }

    err = con.QueryOne(c, `
        INSERT Movie {
            title := <str>$0,
            director := (SELECT Director FILTER .id = <uuid>$1 LIMIT 1),
        }`,
        &result,
        payload.Title,
        payload.Director.ID,
    )

    ...
}
```

Reply:
Ah, got it, thanks for explaining!